test/write.test.js

```diff
@@ -95,3 +159,126 @@ test('Writing - All in one big test', async (t) => {
 
   t.end();
 });
+
+test('Writing - from custom stream', async (t) => {
+  const { readable, writable } = new Writer();
+  const inputChunks = [[1, 2, 3, 4], [5, 6, 7, 8]];
+
+  const inputReadable = getInputReadable({ inputChunks })
+  const writer = writeStuff(writable, inputReadable);
+
+  const outputWritable = getOutputWritable();
+  await readable.pipeTo(outputWritable);
+
+  // Extract stored chunks off custom property on stream object
+  let { outputChunks } = outputWritable;
+
+  // Remove header and footer
+  outputChunks = outputChunks.slice(1, inputChunks.length + 1);
+
+  t.deepEqual(outputChunks, inputChunks);
+});
+
+test('Writing - abort input readable stream', async (t) => {
+  const { readable, writable } = new Writer();
+
+  const inputReadable = getInputReadable({ abort: true });
+  const writer = writeStuff(writable, inputReadable);
+
+  const outputWritable = getOutputWritable();
+
+  try {
+    await readable.pipeTo(outputWritable);
+  } catch(error) {
+    t.equals(error, ABORT_MESSAGE);
+  }
+});
+
+test('Writing - cancel input readable stream', async (t) => {
+  const { readable, writable } = new Writer();
+
+  const inputReadable = getInputReadable();
+  inputReadable.cancel(ABORT_MESSAGE);
+  const writer = writeStuff(writable, inputReadable);
+
+  const outputWritable = getOutputWritable();
+
+  try {
+    await readable.pipeTo(outputWritable);
+  } catch(error) {
+    const { cancelReason } = inputReadable;
+    const { abortReason } = outputWritable;
+
+    t.equals(error, ABORT_MESSAGE);
+    t.equals(abortReason, ABORT_MESSAGE);
+    t.equals(cancelReason, ABORT_MESSAGE);
+  }
+});
+
+test('Writing - abort output writable stream', async (t) => {
+  const { readable, writable } = new Writer();
+
+  const inputReadable = getInputReadable();
+  const writer = writeStuff(writable, inputReadable);
+
+  const outputWritable = getOutputWritable({ long: true });
+
+  try {
+    const promise = readable.pipeTo(outputWritable);
```

Comment:
```suggestion
    const controller = new AbortController();
    const promise = readable.pipeTo(outputWritable, { signal });
    controller.abort();
```

I think our abortion semantics should use `AbortController`, so you can pass in `signal` here.